zebrad/src/components/mempool/error.rs

```diff
@@ -0,0 +1,24 @@
+//! Errors that can occur when adding transactions to the mempool or querying it.
+
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum TransactionInsertError {
```

Comment:
```suggestion
#[allow(dead_code)]
pub enum TransactionInsertError {
```

Should remove the never constructed warnings.